### ISSUE TITLE
[feature] Disable auto-expand for unreadable repositories

### ISSUE THREAD

**Issue by NJAldwin** [EnixCoda/Gitako#128]
I'd like the option to leave 'Auto expand' enabled, and have the sidebar not expand if the repo cannot be read.

Use case: I cannot use Gitako for my organization's private repos (too risky), but I would like to use it for public repos.  My current options are to either:
- disable auto-expand and always expand manually on public repos
- enable auto-expand and always hide manually on private repos

Adding the ability to auto-hide if there are no permissions to read files in the current repo would make this a bit nicer to use :) 💅🏼 

(apologies if this feature request already exists -- I searched and didn't see any relevant ones)

**Comment by EnixCoda** [comment]
Hi @NJAldwin , thank you for the feedback!

It's a good suggestion. I'm worried that if hide on all no-permissions situations would make new users confused. Asking tokens from users eagerly should make it easier & faster for them to pick up Gitako. What about adjusting the auto-expand logic to this:

```
if requests failed due to authorization
  if a token is set
    hide // the user is aware of the concept of token and chose to deny Gitako from accessing current repo
  else
    show
```

**Comment by NJAldwin** [comment]
Sure, that seems totally reasonable.  I had been thinking of making it another option (always expand by default, so it wouldn't confuse new users like you mentioned).  Either would be fine with me!  Thanks.

**Comment by EnixCoda** [comment]
Fixed, will be included in the next release, `v2.8`.
